### PR TITLE
Upgrading calico and specs

### PR DESCRIPTION
**Pull request by knabben**
Upgrading to Calico 3.23 latest
Fixing issue with Kubernetes 1.24 and removed parameters in the StartKubelet.ps1

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -78,7 +78,7 @@ Vagrant.configure(2) do |config|
       if not File.file?(".lock/cni") then
         if cni == "calico" then
           # installs both felix and node
-          winw1.vm.provision "shell", path: "sync/windows/0-calico.ps1", privileged: true
+          winw1.vm.provision "shell", path: "sync/windows/0-calico.ps1", args: "#{kubernetes_version}", privileged: true
```
Comment (knabben) [inline Vagrantfile:81]: passing version to calico installation

File: fetch.sh [hunk 2]
```diff
@@ -29,15 +35,17 @@ if [ ! -z ${kubernetes_sha} ]; then
...
-  echo "kubernetes/ exists, not cloning..."
-  pushd kubernetes
-    git checkout $kubernetes_sha -f
-  popd
-else
-  git clone https://github.com/kubernetes/kubernetes.git
-  pushd kubernetes
-    git checkout $kubernetes_sha
-  popd
+if [[ $BUILD_FROM_SOURCE -eq 1 ]]; then
```
Comment (knabben) [inline fetch.sh:38]: cloning kubernetes only when build is from source

File: sync/windows/0-calico.ps1 [hunk 3]
```diff
@@ -13,18 +13,29 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+Param(
+  [parameter(HelpMessage="Kubernetes Version")]
+  [string] $kubernetesVersion=""
+)
 
 # Force Kubernetes folder
 mkdir -Force C:/k/
 
+# Copy a clean StartKubelet.ps1 configuration for 1.24+
+If ([int]$kubernetesVersion.split(".",2)[1] -gte 24) {
```
Comment (knabben) [inline sync/windows/0-calico.ps1:25]: overwrite only when minor is >= 24

**Review by marosset — APPROVED** [review]
/lgtm

File: makefile [hunk 4]
```diff
@@ -28,7 +28,7 @@ all: 0-fetch-k8s 1-build-binaries 2-vagrant-up 3-smoke-test 4-e2e-test
 
 1-build-binaries:
 	@chmod +x build.sh
-	@./build.sh $(path)
+	@./build.sh $(PWD)/kubernetes
```
Comment (aravindhp) [inline makefile:31]: This breaks specifying `make path=\kubedir 1` i.e. to the fork where the dev work is being done.

**Pull request merged**